fvm/src/gas/price_list.rs

```diff
@@ -860,6 +860,15 @@ impl PriceList {
         self.wasm_rules.memory_expansion_per_byte_cost * grow_memory_bytes as i64
     }
 
+    /// Returns the gas required for initializing tables.
+    pub fn init_table_gas(&self, min_table_elements: u32) -> Gas {
+        // Each element reserves a `usize` in the table, so we charge 8 bytes per pointer.
+        // https://docs.rs/wasmtime/2.0.2/wasmtime/struct.InstanceLimits.html#structfield.table_elements
+        self.wasm_rules.memory_expansion_per_byte_cost
+            * (min_table_elements as i32)
```

Comment:
Let's deal with overflow here.

Reply:
Doesn't the way multiplication is implemented deal with it? 

https://github.com/filecoin-project/ref-fvm/blob/master/fvm/src/gas/mod.rs#L141

Reply:
Ah, I mean wrap around when casting to an i32.

Reply:
Oh, good catch, that could have been a nasty vulnerability :facepalm: 

I added variants of `Gas::mul` to handle these so we don't have to keep casting.